### DISCUSSION TITLE
Make mantine packages play nice with nextjs optimizePackageImports.

### DISCUSSION THREAD

**Discussion by Abhirup-99** [mantinedev/mantine#7007] [Feature requests]
This really reduces load times, but I am getting 404 on some packages. So I don't think this will be a lot of work, but just need to be looked into. Thanks for the library, really appreciate the effort.

**Comment by rtivital** [comment]
There is an example template with app router, it works correctly – https://github.com/mantinedev/next-app-template/blob/master/next.config.mjs#L13
pages router does not need this feature, tree shaking works correctly out of the box

**Comment by Abhirup-99** [comment]
@rtivital Just FYI, https://github.com/vercel/next.js/tree/canary/test/development/basic/barrel-optimization/fixture/pages
pages router does need the optimizePackageImports. I had some issues with CSS getting 404, but it went away on clearing the .next folder.

**Comment by Abhirup-99** [comment]
Pointer, getting ` GET /_next/static/chunks/pages/UnstyledButton.module.css.mjs.map 404`
after adding the `optimizePackageImports` option.
